### PR TITLE
fix(auth): Change auth to service account

### PR DESCRIPTION
**Pull request by shaharyair**
changing to service account for auth without contest page. for google api to work on docker container

### CODE REVIEW

File: drive_logic.py [hunk 1]
```diff
@@ -4,15 +4,13 @@
...
-from google.oauth2.credentials import Credentials
-from google_auth_oauthlib.flow import InstalledAppFlow
+from google.oauth2 import service_account
 from googleapiclient.discovery import build
 from googleapiclient.errors import HttpError
-from google.auth.exceptions import RefreshError
 from googleapiclient.http import MediaIoBaseDownload
 
-DOWNLOADED_STORIES_DIR = os.path.join(os.path.abspath(os.curdir), 'DownloadedStories')
+DOWNLOADED_STORIES_DIR = os.path.join(
```
Comment (shakos1) [inline drive_logic.py:12]: New line changes belong to 'ref' commit

File: drive_logic.py [hunk 2]
```diff
@@ -33,6 +32,7 @@ class DriveError(HttpError):
 
 class DriveDownloadError(DriveError):
     """Raised when encountered an error while downloading from Drive"""
+
```
Comment (shakos1) [inline drive_logic.py:35]: New line changes belong to 'ref' commit

File: drive_logic.py [hunk 3]
```diff
@@ -43,6 +43,7 @@ def __str__(self):
 
 class DriveMultipleFolders(DriveError):
     """Raised when found multiple folders for specified location-folder."""
+
```
Comment (shakos1) [inline drive_logic.py:46]: New line changes belong to 'ref' commit

File: drive_logic.py [hunk 4]
```diff
@@ -53,6 +54,7 @@ def __str__(self):
 
 class DriveLocationNotFound(DriveError):
     """Raised when no folder was found for specified location-folder."""
+
```
Comment (shakos1) [inline drive_logic.py:57]: New line changes belong to 'ref' commit

File: drive_logic.py [hunk 5]
```diff
@@ -139,14 +125,16 @@ def get_files_at_date_in_folder(self,
                 if not items:
                     logger.info('No files found.')
                     return []
-                files.extend([{"id": item["id"], "name": item["name"]} for item in items])
+                files.extend([{"id": item["id"], "name": item["name"]}
```
Comment (shakos1) [inline drive_logic.py:128]: New line changes belong to 'ref' commit

File: drive_logic.py [hunk 6]
```diff
@@ -139,14 +125,16 @@ def get_files_at_date_in_folder(self,
...
+                             for item in items])
                 page_token = results.get('nextPageToken', None)
                 if page_token is None:
                     break
 
         except HttpError as error:
             logger.error(f"An error occurred: {error}")
-            raise HttpError(resp=error.resp, content=error.content, uri=error.uri)
+            raise HttpError(resp=error.resp,
```
Comment (shakos1) [inline drive_logic.py:136]: New line changes belong to 'ref' commit

File: drive_logic.py [hunk 7]
```diff
@@ -159,8 +147,10 @@ def get_files(self, location: str,
         :return: list of files as dictionaries - [{id: ..., name: ...}, {id: ..., name: ...}, ... ]
         :exception: HttpError: Couldn't get files from Drive
         """
-        start_date = datetime.datetime(year=start_year, month=start_month, day=start_day)
-        end_date = datetime.datetime(year=end_year, month=end_month, day=end_day)
+        start_date = datetime.datetime(
```
Comment (shakos1) [inline drive_logic.py:150]: New line changes belong to 'ref' commit

File: drive_logic.py [hunk 8]
```diff
@@ -197,7 +187,8 @@ def _download(self, file_id: int, file_name: str) -> str:
         try:
             while not done:
                 status, done = downloader.next_chunk()
-                logger.debug(f"Download status: {int(status.progress() * 100)}")
+                logger.debug(
```
Comment (shakos1) [inline drive_logic.py:189]: New line changes belong to 'ref' commit

File: drive_logic.py [hunk 9]
```diff
@@ -206,7 +197,8 @@ def _download(self, file_id: int, file_name: str) -> str:
             with open(file_path, "wb") as f:
                 shutil.copyfileobj(fh, f)
 
-            logger.success(f"Downloaded {file_name} and saved it in {file_path}")
+            logger.success(
```
Comment (shakos1) [inline drive_logic.py:199]: New line changes belong to 'ref' commit

File: drive_logic.py [hunk 10]
```diff
@@ -220,15 +212,17 @@ def download_files(self, location: str, start_year: int, start_month: int, start
                                      end_year=end_year, end_month=end_month, end_day=end_day)
         downloaded_files = []
         for file in drive_files:
-            downloaded_files.append({'id': file['id'], 'path': self._download(file['id'], file['name'])})
+            downloaded_files.append(
+                {'id': file['id'], 'path': self._download(file['id'], file['name'])})
```
Comment (shakos1) [inline drive_logic.py:215]: New line changes belong to 'ref' commit

File: drive_logic.py [hunk 11]
```diff
@@ -220,15 +212,17 @@ def download_files(self, location: str, start_year: int, start_month: int, start
...
 
         return downloaded_files
 
     def get_download_link(self, file_id: str):
         """Get the url link to download the file from drive"""
         logger.debug('Getting download link...')
         service = build(API_NAME, API_VERSION, credentials=self.creds)
-        file_metadata = service.files().get(fileId=file_id, fields='webContentLink').execute()
+        file_metadata = service.files().get(
```
Comment (shakos1) [inline drive_logic.py]: New line changes belong to 'ref' commit

File: drive_logic.py [hunk 12]
```diff
@@ -294,7 +288,8 @@ def _get_locations(self, dashboard: str) -> list:
 
         except HttpError as error:
             logger.error(f"An error occurred: {error}")
-            raise HttpError(resp=error.resp, content=error.content, uri=error.uri)
+            raise HttpError(resp=error.resp,
```
Comment (shakos1) [inline drive_logic.py:286]: New line changes belong to 'ref' commit

File: drive_logic.py [hunk 13]
```diff
@@ -309,6 +304,7 @@ def get_locations_and_dates(self, dashboard: str) -> list:
         for location_folder in locations_folders:
             location_name = location_folder.split('_')[0]
             location_dates = self.get_location_dates(location_folder)
-            locations_and_dates.append({'name': location_name, 'location_dates': location_dates})
+            locations_and_dates.append(
```
Comment (shakos1) [inline drive_logic.py:302]: New line changes belong to 'ref' commit

**Review by shakos1 — APPROVED** [review]
Move new line changes to a new commit and consider it approved 👍

File: drive_logic.py (hunk 1 above)
Comment (shaharyair) [reply drive_logic.py:12]: added changes to descriptions

File: drive_logic.py (hunk 2 above)
Comment (shaharyair) [reply drive_logic.py:46]: added changes to description

File: drive_logic.py (hunk 3 above)
Comment (shaharyair) [reply drive_logic.py:57]: added changes to description

File: drive_logic.py (hunk 4 above)
Comment (shaharyair) [reply drive_logic.py:68]: added changes to description

File: drive_logic.py [hunk 14]
```diff
@@ -107,7 +92,8 @@ def get_location_directory(self, location: str) -> str:
             location_directory = []
             for folder in folders:
                 location_directory.append(folder['id'])
-                logger.debug(f'Found folder with the name: {folder["name"]} and the ID: {folder["id"]}')
+                logger.debug(
```
[the remark this answers is not in the document]
Comment (shaharyair) [reply drive_logic.py:120]: added changes to description

File: drive_logic.py (hunk 5 above)
Comment (shaharyair) [reply drive_logic.py:152]: added changes to description

File: drive_logic.py (hunk 6 above)
Comment (shaharyair) [reply drive_logic.py:160]: added changes to description

File: drive_logic.py (hunk 7 above)
Comment (shaharyair) [reply drive_logic.py:174]: added changes to description

File: drive_logic.py (hunk 9 above)
Comment (shaharyair) [reply drive_logic.py:223]: added changes to description

File: drive_logic.py (hunk 10 above)
Comment (shaharyair) [reply drive_logic.py:239]: added changes to description

File: drive_logic.py (hunk 11 above)
Comment (shaharyair) [reply drive_logic.py]: added changes to description

File: drive_logic.py (hunk 12 above)
Comment (shaharyair) [reply drive_logic.py:310]: added changes to description

File: drive_logic.py (hunk 13 above)
Comment (shaharyair) [reply drive_logic.py:326]: added changes to description

**Pull request merged**